src/pages/Earn/useFarmRegistry.ts

```diff
@@ -89,3 +94,17 @@ export const useFarmRegistry = () => {
 
   return farmSummaries
 }
+
+export const useHigherFarmSummaries = () => {
+  const farmSummaries = useFarmRegistry()
+
+  return farmSummaries.reduce((prev: FarmSummary[], current) => {
```

Comment:
There's a simpler way to write this. Also, don't use `farmName` as the key. It may not be sanitized properly from the backend.


```
const farmSummaries = useFarmRegistry()

const farmsUniqueByBestFarm farmSummaries.reduce((prev, curr) => {
  if (!prev[curr.stakingToken]) {
    prev[curr.stakingToken] = curr;
  } else if(curr.rewardsUSDPerYear > prev[curr.stakingToken].rewardsUSDPerYear) {
    prev[curr.stakingToken] = curr;
  }
  return prev;
}, {})

return Object.values(farmsUniqueByBestFarm);
```

Reply:
![image](https://user-images.githubusercontent.com/95660534/146805682-726f56ef-1213-4a06-ae98-ad7f8321abf2.png)
I modified the logic as you mentioned.